src/query/plan/rule_based_planner.cpp

```diff
@@ -658,32 +664,34 @@ std::unique_ptr<LogicalOperator> GenNamedPaths(std::unique_ptr<LogicalOperator>
 std::unique_ptr<LogicalOperator> GenReturn(Return &ret, std::unique_ptr<LogicalOperator> input_op,
                                            SymbolTable &symbol_table, bool is_write,
                                            const std::unordered_set<Symbol> &bound_symbols, AstStorage &storage,
-                                           PatternComprehensionDataMap &pc_ops) {
+                                           PatternComprehensionDataMap &pc_ops, Expression *commit_frequency) {
   // Similar to WITH clause, but we want to accumulate when the query writes to
   // the database. This way we handle the case when we want to return
   // expressions with the latest updated results. For example, `MATCH (n) -- ()
   // SET n.prop = n.prop + 1 RETURN n.prop`. If we match same `n` multiple 'k'
   // times, we want to return 'k' results where the property value is the same,
   // final result of 'k' increments.
-  bool accumulate = is_write;
+  bool const has_periodic_commit = !!commit_frequency;
+  bool const accumulate = is_write && !has_periodic_commit;
```

Comment:
did we discuss this already on the call? is this non-issue?
since periodic commit needs to remove the accumulates so it can pass the operator necessary number of times and in the end free deltas

Reply:
yes, we did this, sorry, comment was left before call, so I didn't remove it